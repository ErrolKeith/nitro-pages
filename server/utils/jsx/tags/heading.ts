import { makeJsx } from "./makeJsx";

export type HeadingLevel = 1 | 2 | 3 | 4 | 5 | 6;

export interface HeadingProps {
  level: HeadingLevel;
  text: string;
}

export function makeHeading(config: HeadingProps) {
  return makeJsx(`h${config.level}`, {}, config.text);
}