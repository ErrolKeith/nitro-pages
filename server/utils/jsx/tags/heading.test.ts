import { it, expect } from "vitest";
import { makeHeading, HeadingProps } from "./heading";

it("can make heading tags", () => {
  const headingTestProps: HeadingProps = {
    level: 1,
    text: "Sample heading level 1",
  };
  const headingTest = makeHeading(headingTestProps);
  expect(headingTest).toMatchSnapshot(headingTest);
});
