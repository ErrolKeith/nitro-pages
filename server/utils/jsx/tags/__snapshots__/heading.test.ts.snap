// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can make heading tags 1`] = `
<h1>
  Sample heading level 1
</h1>
`;
